### PR TITLE
feat: Portal 1138/pill component

### PR DESCRIPTION
**Pull request by cfarmer-fearless**
This is a new component following this figma design: https://www.figma.com/file/TrW6WMf6GYgscPkVQaiMZ0/CDC-React-Components?node-id=1762%3A6810&mode=dev

Still TODO:
- Inverse colors (done)
- Use Sass vars for colors (done)
- Props for text/icon color and background color (done)
- Aria label prop (is this different from SR text prop?) (done)

### CODE REVIEW

File: packages/cdc-react/src/components/Pill/Pill.test.tsx [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+    render(<Pill label="hello" />);
+    expect(screen.getByText("hello")).toBeInTheDocument();
+  });
+
+  it("should truncate long text", () => {
+    render(<Pill label="This is a long pull that should be truncated" />);
+    expect(
+      screen.getByTitle("This is a long pull that should be truncated")
+    ).toBeInTheDocument();
+  });
```
Comment (sublettk) [inline packages/cdc-react/src/components/Pill/Pill.test.tsx:15]: This test seems to only be testing if the full text is being displayed, not if it is truncated. To test if it is truncated, try adjusting the text to the truncated version and see if it is being displayed.

File: packages/storybook/src/stories/Pill.stories.tsx
Comment (sublettk) [inline packages/storybook/src/stories/Pill.stories.tsx:1]: Consider adding a story that is similar to the "Variations" section of the Figma file. The variations are more aligned with our intended project goals for creating this component.

File: packages/cdc-react/src/components/Pill/Pill.tsx [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+const TRUNCATE_LENGTH = 27;
+
+export const pillVariations = ["default", "toggle", "input", "info"];
+type PillVariation = (typeof pillVariations)[number];
+
+export const pillShapes = ["slot", "roundCorners"];
+type PillShape = (typeof pillShapes)[number];
+
+export type PillProps = (
+  | {
```
Comment (sublettk) [inline packages/cdc-react/src/components/Pill/Pill.tsx:15]: Look at you with your advanced TypeScript! Nice work!

File: packages/cdc-react/src/components/Pill/Pill.tsx [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+
+    return (
+      <button
+        className={`${classes} ${isToggled ? "active" : ""}`}
+        title={showTitle ? props.label : undefined}
+        onClick={handleToggle}>
+        <span className="label truncate">{props.label}</span>
+      </button>
+    );
+  }
```
Comment (sublettk) [inline packages/cdc-react/src/components/Pill/Pill.tsx:96]: Could you make these into ternary ops?

File: packages/storybook/src/stories/Pill.stories.tsx
Comment (cfarmer-fearless) [reply packages/storybook/src/stories/Pill.stories.tsx:1]: For sure!  I'm going to be adding a `color` prop and defining some `success`, `warning`, `error` type of color variables that map to some named color types.

File: packages/cdc-react/src/components/Pill/Pill.test.tsx (hunk 1 above)
Comment (sublettk) [reply packages/cdc-react/src/components/Pill/Pill.test.tsx:15]: I think that checking for the class to be present is a good idea.

Comment (cfarmer-fearless) [reply packages/cdc-react/src/components/Pill/Pill.test.tsx:15]: Done

File: packages/cdc-react/src/components/Pill/Pill.tsx (hunk 3 above)
Comment (cfarmer-fearless) [reply packages/cdc-react/src/components/Pill/Pill.tsx]: Done

**Review by ian-sawyer — APPROVED** [review]
LGTM!

**Review by sublettk — APPROVED** [review]
Nice work on this component! I love seeing more tests going into our components and maximizing the use of TypeScript in ways that makes our code more robust.

**Pull request merged**